taca_ngi_pipeline/utils/nbis_xml_generator.py

```diff
@@ -0,0 +1,304 @@
+                                      '\t\t\t\t<LIBRARY_LAYOUT>{layout}</LIBRARY_LAYOUT>\n'
+                                      '\t\t\t\t<LIBRARY_CONSTRUCTION_PROTOCOL>{protocol}</LIBRARY_CONSTRUCTION_PROTOCOL>\n'
+                                      '\t\t\t</LIBRARY_DESCRIPTOR>\n'
+                                      '\t\t</DESIGN>\n'
+                                      '\t\t<PLATFORM>\n'
+                                      '\t\t\t<ILLUMINA>\n'
+                                      '\t\t\t\t<INSTRUMENT_MODEL>{instrument}</INSTRUMENT_MODEL>\n'
+                                      '\t\t\t</ILLUMINA>\n'
+                                      '\t\t</PLATFORM>\n'
+                                      '\t\t<PROCESSING/>\n'
+                                      '\t</EXPERIMENT>\n').format(**sample_stat['experiment'])
+            # fill in to run values from collected stat
+            run_xml_string += ('\t<RUN alias="{alias}" run_center="National Genomics Infrastructure, Stockholm" center_name="">\n'
+                               '\t\t<EXPERIMENT_REF refname="{exp_ref}"/>\n'
+                               '\t\t<DATA_BLOCK>\n'
+                               '\t\t\t<FILES>\n'
+                               '{files}'
+                               '\t\t\t</FILES>\n'
+                               '\t\t</DATA_BLOCK>\n'
+                               '\t</RUN>\n').format(**sample_stat['run'])
+        # wrap in final xml string tags
+        experiment_set = ('<EXPERIMENT_SET xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" '
+                          'xsi:noNamespaceSchemaLocation="ftp://ftp.sra.ebi.ac.uk/meta/xsd/sra_1_5/SRA.experiment.xsd">'
+                          '\n{}\n</EXPERIMENT_SET>\n').format(experiment_xml_string)
+        run_set = ('<RUN_SET  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" '
+                   'xsi:noNamespaceSchemaLocation="ftp://ftp.sra.ebi.ac.uk/meta/xsd/sra_1_5/SRA.run.xsd">'
+                   '\n{}\n</RUN_SET>\n').format(run_xml_string)
+        # dont save in file if asked to return as string
+        if return_string_dict:
+            return {'experiments': experiment_set, 'runs': run_set}
+        # save in files in given outdir
+        with open(os.path.join(self.outdir, "{}_experiments.xml".format(self.project['project_id'])), 'w') as exml, \
+             open(os.path.join(self.outdir, "{}_runs.xml".format(self.project['project_id'])), 'w') as rxml:
+            exml.write(experiment_set)
+            rxml.write(run_set)
+
+
+    def _collect_sample_stats(self):
+        """ Collect stats that will be used to generate the xml files """
+        for sample in sorted(self.samples_delivered.keys()):
+            # all the samples should exist, if not fail right away
+            sample_seq_instrument = self.sample_aggregated_stat[sample]
+            for inst in sorted(sample_seq_instrument.keys()):
```

Comment:
Just for readability, maybe consider changing `inst` to `instrument`? It's pretty clear that it's short for instrument here, but it could also be interpreted as instance.

Reply:
Done!